src/link/mod.rs

```diff
@@ -164,10 +161,10 @@ impl Import {
         let name = name.to_string();
         let unique_id = Import::unique_id(&path, &name);
         Import {
+            id: Some(StringId::new(path.clone(), name.clone())),
```

Comment:
I felt this would be better to put in a follow up PR to keep the scale of changes down. In general I find it is easier to merge multiple smaller changes than a single very large change. I recognize there are some benefits to changing it here, but I am not convinced they necessarily are worth the additional size of the PR.